ahcore/writers.py

```diff
@@ -54,12 +83,22 @@ def __init__(
         self._logger = logger  # maybe not the best way, think about it
         self._logger.debug("Writing h5 to %s", self._filename)
 
-    def init_writer(self, first_batch: GenericArray, h5file: h5py.File) -> None:
+    def init_writer(self, first_coordinates: GenericArray, first_batch: GenericArray, h5file: h5py.File) -> None:
         """Initializes the image_dataset based on the first tile."""
-        batch_shape = np.asarray(first_batch).shape
-        batch_dtype = np.asarray(first_batch).dtype
+
+        if self._is_compressed_image:
+            # We need to read the first batch as it is a compressed PIL image
+            _first_pil = decode_array_to_pil(first_batch[0])
+            _mode = _first_pil.mode
+            _format = _first_pil.format
+            _num_channels = len(_first_pil.getbands())
+        else:
+            _mode = "ARRAY"
+            _format = "RAW"
+            _num_channels = first_batch.shape[-1]
```

Comment:
I think it's correct as it's the conversion of a PIL Image to an array. If it would be a tensor it would indeed be channels first:

```python
import PIL.Image
import numpy as np
print(np.asarray(PIL.Image.open("image.jpg")).shape)

(1635, 1966, 3)
```